plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/N4JSResourceValidator.java

```diff
@@ -41,6 +46,37 @@
 	@Inject
 	private OperationCanceledManager operationCanceledManager;
 
+	@Override
+	public List<Issue> validate(Resource resource, CheckMode mode, CancelIndicator cancelIndicator) {
+		if (resource instanceof N4JSResource) {
+			final N4JSResource resourceCasted = (N4JSResource) resource;
+			try {
+				resourceCasted.performPostProcessing(cancelIndicator);
+			} catch (Throwable th) {
+				// ignore this exception/error (we will create an issue for it below)
```

Comment:
You could also store the exception in the `catch` via a local variable here and do not need an extra field for `postProcessingThrowable ` in `PostProcessingAwareResource.java`

Reply:
No, this isn't possible. In the standard case, post-processing will already have taken place when method `#validate()` is entered.